hm_pyhelper/miner_param.py

```diff
@@ -52,6 +52,31 @@ def get_gateway_mfr_test_result():
     return False
 
 
+def provision_key():
+    """
+    Attempt to provision key.
+    """
+    direct_path = os.path.dirname(os.path.abspath(__file__))
+    gateway_mfr_path = os.path.join(direct_path, 'gateway_mfr')
+
+    if get_gateway_mfr_test_result():
```

Comment:
This kind of looks like there will be false positives when a key has not yet been provisioned, but like you pointed out this is hard to test without a fresh device. 

My thinking is that [get_gateway_mfr_tes_result](https://github.com/NebraLtd/hm-pyhelper/blob/cdcc658d342f66c9b0f3b8f9028eb130102d49d3/hm_pyhelper/miner_param.py#L49) will return true whenever the JSON returned by `gateway_mfr test` can be parsed

I'm not 100% but I think you may want to check the test response specifically for the `"test": "miner_key(0)"` in the corresponding `result` field.

```
{
  "result": "pass",
  "tests": [
    {
      "output": "ok",
      "result": "pass",
      "test": "serial"
    },
    {
      "output": "ok",
      "result": "pass",
      "test": "zone_locked(data)"
    },
    {
      "output": "ok",
      "result": "pass",
      "test": "zone_locked(config)"
    },
    {
      "output": "ok",
      "result": "pass",
      "test": "slot_config(0..=15, ecc)"
    },
    {
      "output": "ok",
      "result": "pass",
      "test": "key_config(0..=15, ecc)"
    },
    {
      "output": "ok",
      "result": "pass",
      "test": "miner_key(0)"
    }
  ]
}
```

The missing link is figuring out what [compact_key_in_slot](https://github.com/helium/gateway-mfr-rs/blob/main/src/tests.rs#L162) does exactly. Sounds self explanatory.

Reply:
> My thinking is that get_gateway_mfr_tes_result will return true whenever the JSON returned by gateway_mfr test can be parsed

Good catch. Yes this is correct. Pushed an update.